### PR TITLE
refactor: MachineListTable

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- refactor: MachineListTable
  - extract tableModels to own file

### QA steps

- Go to machine list
- Verify the machine list loaded and pagination, grouping and row selection works as before.

### CODE REVIEW

File: src/app/machines/views/MachineList/MachineListTable/types.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import type { SortDirection } from "app/base/types";
+import type { MachineColumns } from "app/machines/constants";
+import type { GetMachineMenuToggleHandler } from "app/machines/types";
+import type {
+  Machine,
+  MachineStateListGroup,
+  FetchGroupKey,
+} from "app/store/machine/types";
+
+export type Props = {
```
Comment (Jay-Topher) [inline src/app/machines/views/MachineList/MachineListTable/types.ts:15]: I think naming the type `Props` was okay when it was a local type, it looks too generic now that it's being shared between files and can be mistaken for another prop type. We could call it `MachineListTableProps`

Comment (ndv99) [reply src/app/machines/views/MachineList/MachineListTable/types.ts:15]: Agreed, marking this as `Code +1 (with changes)`

Comment (ndv99) [inline src/app/machines/views/MachineList/MachineListTable/types.ts:15]:
```suggestion
export type MachineListTableProps = {
```

**Review by ndv99 — CHANGES_REQUESTED** [review]

File: src/app/machines/views/MachineList/MachineListTable/types.ts (hunk 1 above)
Comment (petermakowski) [reply src/app/machines/views/MachineList/MachineListTable/types.ts:15]: Good point, will update!

**Review by ndv99 — APPROVED** [review]

**Pull request merged**